src/net.cpp

```diff
@@ -1814,6 +1814,16 @@ void CConnman::CreateNodeFromAcceptedSocket(std::unique_ptr<Sock>&& sock,
     }
 
     const bool inbound_onion = std::find(m_onion_binds.begin(), m_onion_binds.end(), addr_bind) != m_onion_binds.end();
+
+    // Don't accept inbound connections from peers already connected to us.
+    // Skip this check for Tor, as their inbound/outbound addresses differ.
+    if (!inbound_onion) {
+        if (auto pnode = AlreadyConnectedToAddress(addr)) {
```

Comment:
Similar question here: Could this make some local network testing setups no longer viable?